app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt

```diff
@@ -64,353 +42,143 @@ import androidx.compose.ui.text.UrlAnnotation
-            remember(context) {
-              val imageBitmap =
-                (AppCompatResources.getDrawable(context, R.mipmap.ic_launcher)
-                    as AdaptiveIconDrawable)
-                  .toBitmap()
-                  .asImageBitmap()
-              BitmapPainter(imageBitmap)
+              IntOffset(
+                x = navButtonWidthPx - titleCoordinates.x.roundToInt(),
+                y = targetY + parallaxOffset.roundToInt(),
+              )
+            } else {
+              IntOffset.Zero
             }
-          }
-        Image(
-          painter = icon,
-          contentDescription = "CatchUp icon",
-          modifier =
-            Modifier.size(48.dp)
-              .align(CenterHorizontally)
-              .graphicsLayer { alpha = bannerAlpha }
-              .clickable { debugUiCount++ }
-        )
-        Spacer(Modifier.height(8.dp))
-        var fixedTitleHeight by remember { mutableStateOf<Dp?>(null) }
-        val heightModifier = fixedTitleHeight?.let { Modifier.requiredHeight(it) } ?: Modifier
-        val density = LocalDensity.current
-        val titleDebugBackgroundBottom = if (debugUi) Modifier.background(Color.Red) else Modifier
-        val titleDebugBackgroundTop = if (debugUi) Modifier.background(Color.Green) else Modifier
-        Text(
-          modifier =
-            Modifier.align(CenterHorizontally)
-              .then(titleDebugBackgroundBottom)
-              .offset { offset }
-              .onGloballyPositioned { coordinates ->
-                titleHeight = coordinates.size.height
-                actualY = coordinates.positionInParent().y
-                if (debugUi) {
-                  println(
-                    "ZAC: y: ${coordinates.positionInParent().y} - ${coordinates.positionInParent().y} - ${coordinates.positionInWindow().y}"
-                  )
-                }
-                if (!titleLaidOut) {
-                  fixedTitleHeight = density.run { coordinates.size.height.toDp() }
-                  if (debugUi) {
-                    println("ZAC: fixedTitleHeight: $fixedTitleHeight, pinnedHeight: $pinnedHeight")
-                  }
-
-                  // positionInParent calls do not account for the system bar
-                  titleY = coordinates.positionInParent().y
 
-                  // TODO account for parallax effect of shrinking column
-                  // TODO account for system bar
-                  val settledTitleOffset =
-                    IntOffset(
-                      x = coordinates.positionInParent().x.roundToInt(),
-                      y = coordinates.positionInParent().y.roundToInt()
-                    )
+          Text(
+            modifier =
+              Modifier.align(CenterHorizontally)
+                .onGloballyPositioned { titleCoordinates = it.positionInParent() }
+                .offset {
+                  lerp(IntOffset.Zero, textOffset, 1f - scaffoldState.toolbarState.progress)
```

Comment:
I tried using `arcLerp` here, but it's producing a terrible result.

Reply:
it's extremely scrapped together so not surprising, I'm not sure I did it right